Dockerfile

```diff
@@ -4,34 +4,35 @@ FROM golang:alpine as builder
 # queries required to connect to linked containers succeed.
 ENV GODEBUG netdns=cgo
 
-# Install dependencies and build the binaries.
+# Copy in the local repository to build from.
+COPY . /go/src/github.com/lightningnetwork/lnd
+
+# Install dependencies.
 RUN apk add --no-cache --update alpine-sdk \
     git \
-    make \
-    gcc \
-&&  git clone https://github.com/lightningnetwork/lnd /go/src/github.com/lightningnetwork/lnd \
-&&  cd /go/src/github.com/lightningnetwork/lnd \
-&&  make \
-&&  make install
+    make
+
+# Build the binaries.
+WORKDIR /go/src/github.com/lightningnetwork/lnd
+RUN make install
 
 # Start a new, final image.
 FROM alpine as final
 
 # Define a root volume for data persistence.
 VOLUME /root/.lnd
 
-# Add bash and ca-certs, for quality of life and SSL-related reasons.
-RUN apk --no-cache add \
-    bash \
```

Comment:
I agree bash isn't actually needed to use the image, although as the comment says, it does improve life quality, especially when someone needs to debug the container :)

Reply:
Well, there is already the `/bin/busybox` shell installed by default on alpine, that's enough if you need to debug the container.

Reply:
so no way to use `bash` then?  I guess `busybox` support bourne.

Reply:
There are multiple ways to have `bash` if you really want to install it. You could extend this image, write your own dockerfile, or even install it manually within a running container, but I honestly don't think we need `bash` by default. If that's only because `bash` is "better" (or more practical) than `busybox` then we could apply this argument to a lot of other tools. Docker images are supposed to be as lightweight as possible.

> I guess `busybox` support bourne.

No clue what supporting bourne mean, sorry.